spec/datadog/profiling/collectors/interesting_backtrace_helper.rb

```diff
@@ -0,0 +1,204 @@
+# typed: ignore
+
+# This (very bizarre file) gets used from `stack_spec.rb`. It tries to reproduce the most interesting (contrived?) call
+# stack I can think of, with as many weird variants as possible.
+# The objective is to thoroughly test our stack trace sampling abilities.
+# Control flow goes from bottom of file to top (e.g. ClassA will be the top of the stack)
+
+# ----
+
+class IbhClassA
+  def hello
+    $ibh_ready_queue << true
+    sleep
+  end
+end
+
+module IbhModuleB
+  class IbhClassB < IbhClassA
+    def hello
+      super
+    end
+  end
+end
+
+module IbhModuleC
+  def self.hello
+    IbhModuleB::IbhClassB.new.hello
+  end
+end
+
+class IbhClassWithStaticMethod
+  def self.hello
+    IbhModuleC.hello
+  end
+end
+
+module IbhModuleD
+  def hello
+    IbhClassWithStaticMethod.hello
+  end
+end
+
+class IbhClassC
+  include IbhModuleD
+end
+
+$ibh_a_proc = proc { IbhClassC.new.hello }
```

Comment:
Globals! How about slapping all these `$` bad boys into a module in this file, like `IbhGlobal`?

Reply:
To be clear there, my suggestion will likely accomplish the same level of code isolation, but I just find it jarring to see these `$` around me. It makes me slightly confused when I glance a random `$` while reading code here.

Reply:
Fixed in ce1c971c07012ed2d3c5cee92930417a4b1abd36